extensions/youversion-suggest/src/lookup-reference.tsx

```diff
@@ -1,21 +1,43 @@
 import { List, showToast, Toast } from "@raycast/api";
-import { useCallback, useEffect, useState } from "react";
+import { useCachedPromise } from "@raycast/utils";
+import { useState } from "react";
 import { getReferencesMatchingName } from "youversion-suggest";
 import { getPreferredLanguage, getPreferredVersion } from "./preferences";
 import ReferenceActions from "./reference-actions";
 import { BibleReference } from "./types";
 
 export default function Command() {
-  const { state, search } = useSearch();
+  const [query, setQuery] = useState("");
+
+  const { data: results = [], isLoading } = useCachedPromise(
+    async (searchText: string) => {
+      if (!searchText.trim()) return [];
+      return getReferencesMatchingName(searchText, {
+        language: await getPreferredLanguage(),
+        fallbackVersion: await getPreferredVersion(),
+      });
+    },
+    [query],
+    {
+      keepPreviousData: true,
+      onError: (err) => {
+        showToast({
+          style: Toast.Style.Failure,
+          title: "Could not perform search",
+          message: String(err),
+        });
+      },
```

Comment:
We can simplify this if you only want to change the title:
```suggestion
      failureToastOptions: {
        title: "Could not perform search",
      },
```